includes/class-wc-payments-features.php

```diff
@@ -90,39 +75,6 @@ public static function is_upe_split_enabled() {
 	 * Checks whether the Split UPE with deferred intent creation is enabled
 	 */
 	public static function is_upe_deferred_intent_enabled() {
-		// Support new stores created in 6.4.0 or 6.5.0, and legacy card stores that migrated to deferred UPE in 6.4.0 or 6.5.0.
-		$has_store_enabled_dupe_from_previous_plugin_version = self::has_store_enabled_dupe_from_previous_plugin_version();
-		$has_store_enabled_upe_and_dupe_server_flag          = self::has_store_enabled_upe_and_dupe_server_flag();
-
-		return $has_store_enabled_dupe_from_previous_plugin_version || $has_store_enabled_upe_and_dupe_server_flag;
-	}
-
-	/**
-	 * Checks if the store has deferred intent creation UPE enabled from a previous version.
-	 * This is applicable to:
-	 * * legacy card stores that self-migrated to deferred UPE in 6.4.0 or 6.5.0
-	 * * new stores starting 6.4.0
-	 */
-	private static function has_store_enabled_dupe_from_previous_plugin_version() {
-		return '1' === get_option( self::UPE_DEFERRED_INTENT_FLAG_NAME, '0' );
-	}
-
-	/**
-	 * Checks if the store has UPE enabled and the server-side feature flag is enabled.
-	 * This is applicable to:
-	 * * Split UPE stores starting 6.4.0
-	 * * Legacy UPE stores starting 6.6.0
-	 */
-	private static function has_store_enabled_upe_and_dupe_server_flag() {
-		return ( self::is_upe_split_enabled() || self::is_upe_legacy_enabled() ) && self::is_deferred_upe_server_flag_enabled();
-	}
```

Comment:
This complexity was introduced in https://github.com/Automattic/woocommerce-payments/pull/7306 and was needed for version `6.6.x`. Starting `6.7.0`, every store relies on the server-side flag only, thus this logic can be removed.